rules/DowngradePhp70/Rector/Instanceof_/DowngradeInstanceofThrowableRector.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\DowngradePhp70\Rector\Instanceof_;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\Assign;
+use PhpParser\Node\Expr\BinaryOp\BooleanOr;
+use PhpParser\Node\Expr\Instanceof_;
+use PhpParser\Node\Expr\Variable;
+use PhpParser\Node\Name;
+use PhpParser\Node\Name\FullyQualified;
+use Rector\Core\Rector\AbstractRector;
+use Rector\Naming\Naming\VariableNaming;
+use Rector\NodeTypeResolver\Node\AttributeKey;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * Throwable was introduced in PHP 7.0 so to support older versions we need to also check for Exception.
+ * @see https://www.php.net/manual/en/class.throwable.php
+ * @see \Rector\Tests\DowngradePhp70\Rector\Instanceof_\DowngradeInstanceofThrowableRector\DowngradeInstanceofThrowableRectorTest
+ */
+final class DowngradeInstanceofThrowableRector extends AbstractRector
+{
+    /**
+     * @var string
+     */
+    private const ALREADY_TRANSFORMED = 'downgrade_instanceof_throwable_already_transformed';
+
+    public function __construct(
+        private readonly VariableNaming $variableNaming
+    ) {
+    }
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Add `instanceof Exception` check as a fallback to `instanceof Throwable` to support exception hierarchies in PHP 5',
+            [
+                new CodeSample(
+                    <<<'CODE_SAMPLE'
+return $e instanceof \Throwable;
+CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+return ($throwable = $e) instanceof \Throwable || $throwable instanceof \Exception;
+CODE_SAMPLE
+                ),
+            ]
+        );
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [Instanceof_::class];
+    }
+
+    /**
+     * @param Instanceof_ $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if (! ($node->class instanceof Name && $node->class->toCodeString() === '\\Throwable')) {
+            return null;
+        }
+        $isAlreadyTransformed = (bool) $node->getAttribute(self::ALREADY_TRANSFORMED, false);
```

Comment:
You can use CREATED_BY_RULE attribute that always created on rule applied, eg:

https://github.com/rectorphp/rector-src/blob/34c61ba4498150972db843fca807416540ca71e8/rules/DowngradePhp80/Rector/FuncCall/DowngradeArrayFilterNullableCallbackRector.php#L99-L101